sql/core/src/main/scala/org/apache/spark/sql/json/JSONRelation.scala

```diff
@@ -105,6 +107,15 @@ private[sql] class JSONRelation(
     jsonSchema
   }
 
+  override private[sql] def buildScan(
+      requiredColumns: Array[String],
+      filters: Array[Filter],
+      inputPaths: Array[String],
+      broadcastedConf: Broadcast[SerializableConfiguration]): RDD[Row] = {
+    refresh()
```

Comment:
For json, we do not have other metadata to cache other than file status, right?

`once the table cached` =>  you mean after we call `cache table`? I do see we drop the cached logical plan from our cache manager. Can you try it and see if we drop the cached table data?